### PR TITLE
K-Means & K-Medoids Implementation

### PR DESCRIPTION
**Pull request by rolandhartanto**

### CODE REVIEW

File: rolearnd/clustering/kmeans.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                        if (j == 0):
+                            n_cluster_elmt[k] += 1
+        
+        for cluster_idx in range(0, self.num_clusters):
+            for attr_idx in range(0, data.shape[1]):
+                means[cluster_idx][attr_idx] = sums[cluster_idx][attr_idx] / n_cluster_elmt[cluster_idx]
+
+        return means
+    
+def ndarray_to_dataframe(ndarray):
```
Comment (vincenthend) [inline rolearnd/clustering/kmeans.py:163]: nanti pindahin ke utils aja ya (ga harus sih, apa masukin ke classnya juga ga apa)

File: rolearnd/clustering/kmeans.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        labels = [-1 for i in range(0, len(distances))]
+        for i in range(0, len(distances)):
+            idx, val = self.min_val(distances[i])
+            labels[i] = idx
+        
+        return labels
+        
+    def min_val(self, list_elements):
+        index = -1
+        value = 9999999
```
Comment (vincenthend) [inline rolearnd/clustering/kmeans.py]: ati-ati kalo range data gede
try https://stackoverflow.com/questions/7604966/maximum-and-minimum-values-for-ints

**Review by vincenthend — COMMENTED** [review]
jangan lupa diprivate methodnya rol :v

File: rolearnd/clustering/kmeans.py (hunk 1 above)
Comment (rolandhartanto) [reply rolearnd/clustering/kmeans.py:163]: okay :D

File: rolearnd/clustering/kmeans.py (hunk 2 above)
Comment (rolandhartanto) [reply rolearnd/clustering/kmeans.py]: okay :D

**Pull request merged**